BusinessLogic/Services/TestService.cs

```diff
@@ -42,8 +41,12 @@ public Test CreateTest()
 
     public void SetTest(Test test) => this.test = test;
 
-    public void SaveOrUpdateTest(Test test, bool newTest)
+    public void ProcessTest(Test test, bool newTest, Guid initalId)
```

Comment:
ik zou de acties los houden en deze logica naar de UI verplaatsen. Nu vorm je je service eigenlijk naar de frontend en dat maakt de service weer onnodig complex. (daarnaast misschien single responsibility)

Reply:
Niet eens, nu blijft de logica van het bepalen of iets goed is of niet in de business logica
dus bij het veranderen van de ui blijven dezelfde regels gelden.

Reply:
maar de enige regel om te kijken of iets nieuw is is dat het id nieuw is aangemaakt, dat is denk ik geen business regel

Reply:
overigens word de errorhandeling gedaan in de ui maar het bepalen of iets een error is zijn business rules.

Reply:
ik valideer hier niet of het een nieuwe id is? maar of de targetaudience gewijziged is. Dit is zeker een business rule, wanneer dit gebeurd moet een test automatisch op false komen te staan.

Reply:
maar dat doe je dan toch in de UpdateTest? zitten er geen CRUD methods in het ontwerp

Reply:
wat?

Reply:
deze functie roept de crud functies aan?
update? create?

Reply:
de test op inactief zetten volgens die regel gebeurt alleen als je de test update. waarom doe je die check niet in de UpdateTest method

Reply:
om het single responsibility te houden, dit gebeurd niet in de update functie omdat dat alleen het updaten is. Het processen van de test is dus kijken of alles klopt en of andere business rules nog geapplied moeten worden?

Reply:
Heb het wat voor je aangepast.